cairo/src/order_manager.cairo

```diff
@@ -129,101 +148,135 @@ mod OrderManagerComponent {
+                .order_id_to_order_created
+                .write(id, Order_Created { order_key: order_key_copy, creator: sender });
 
             // Emit event
             self.emit(OrderCreated { id, sender, amount: amount_u128 });
         }
 
         // Execute withdrawal
         fn execute_withdrawal(
-             ref self: ComponentState<TContractState>,
-            message: OrderDetails,
-            token_bridge_helper_address: ContractAddress
+            ref self: ComponentState<TContractState>,
+            message: WithdrawalDetails,
+            token_bridge_helper_address: ContractAddress,
         ) {
             let depositor: EthAddress = message.sender.try_into().unwrap();
-            let order_created = self.order_depositor_to_order_created.read(depositor);
-            
+            let receiver: EthAddress = message.receiver.try_into().unwrap();
+            let order_id: u64 = message.order_id.try_into().unwrap();
+
+            let order_creator = self.get_depositor_from_id(order_id);
+
+            // Verify depositor's identity
+            assert(order_creator == depositor, ERROR_UNAUTHORIZED);
+
+            let order_created = self.order_id_to_order_created.read(order_id);
+
             // Decode order key from stored copy
             let order_key = self.decode_order_key_from_stored_copy(order_created.order_key);
-            let id = order_created.id;
-            
-            // Verify depositor's identity
-            let user = self.get_depositor_from_id(id);
+
             // Check if withdrawal has already been processed
-            let withdrawal_status = self.get_withdrawal_status(depositor, id);
+            let withdrawal_status = self.get_withdrawal_status(depositor, order_id);
 
-            // Ensure the current user is the depositor
-            assert(user == depositor, ERROR_UNAUTHORIZED);
             // Ensure the order has not been withdrawn yet
             assert(!withdrawal_status, ERROR_ALREADY_WITHDRAWN);
 
             let positions = IPositionsDispatcher {
-                contract_address: contract_address_const::<0x06a2aee84bb0ed5dded4384ddd0e40e9c1372b818668375ab8e3ec08807417e5>()
+                contract_address: contract_address_const::<
+                    0x06a2aee84bb0ed5dded4384ddd0e40e9c1372b818668375ab8e3ec08807417e5,
+                >(),
             };
             let this_contract_address = get_contract_address();
-            
-            let amount_sold = positions.withdraw_proceeds_from_sale_to(id, order_key, this_contract_address);
 
-            // Mark as withdrawn
-            self.order_depositor_to_id_to_withdrawal_status.entry(depositor).entry(id).write(true);
+            let amount_sold = positions
+                .withdraw_proceeds_from_sale_to(order_id, order_key, this_contract_address);
+
+            // // Mark as withdrawn
+            self
+                .order_depositor_to_id_to_withdrawal_status
+                .entry(depositor)
```

Comment:
Can this be added to the order_created struct to avoid another mapping? If not all good but I believe we can have it in that struct in order to consolidate